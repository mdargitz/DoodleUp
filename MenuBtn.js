import React from 'react';
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';


class MenuBtn extends React.Component {

  render() {

    let backgroundColor;
    let iconName;
    if (this.props.type === 'Training'){
      backgroundColor = '#BC7C9C';
      iconName = 'line-chart';
    }
    else if (this.props.type === 'LevelUp'){
      backgroundColor = '#A96DA3';
      iconName = 'star';
    }
    else if (this.props.type === 'Inspiration'){
      backgroundColor = '#7A5980';  
      iconName = 'heart';    
    }
    else {
      backgroundColor = '#3B3B58';
      iconName = 'files-o';  
    }

    const styles = StyleSheet.create({
      viewStyle : {

      },
      touchStyle : {
        backgroundColor
      },
      textStyle : {
        color: '#fff',
        fontSize: 50,
        padding: 50
       
      }
    });

    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity
          style={styles.touchStyle}
          onPress={this.props.onPress}
        >
          <Icon name={iconName} size={60} color="#fff" />
          <Text
            style={styles.textStyle}
          >
            {this.props.type}
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}




export default MenuBtn;