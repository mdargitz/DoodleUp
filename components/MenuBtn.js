import React from 'react';
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';


class MenuBtn extends React.Component {

  render() {

    let backgroundColor;
    let iconName;
    if (this.props.type === 'Training'){
      backgroundColor = '#BC7C9C';
      iconName = 'line-chart';
    }
    else if (this.props.type === 'LevelUp'){
      backgroundColor = '#A96DA3';
      iconName = 'star';
    }
    else if (this.props.type === 'Inspiration'){
      backgroundColor = '#7A5980';  
      iconName = 'heart';    
    }
    else if (this.props.type === 'Gesture') {
      backgroundColor = '#A96DA3';
      iconName = 'pencil';
    }
    else if (this.props.type === 'Study'){
      backgroundColor = '#7A5980';  
      iconName = 'eye';
    }
    else if (this.props.type === 'FreeDraw'){
      backgroundColor = '#3B3B58';
      iconName = 'paint-brush'; 
    }
    else {
      backgroundColor = '#3B3B58';
      iconName = 'files-o';  
    }

    const styles = StyleSheet.create({
      viewStyle : {
        margin: 20
      },
      touchStyle : {
        backgroundColor,
        justifyContent: 'center',
        alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.8,
        shadowRadius: 2,
        padding: 30
      },
      textStyle : {
        color: '#fff',
        fontSize: 30,
      },
      subtitleStyle : {
        color: '#fff'
      }

    });

    return (
      <View style={styles.viewStyle}>
        <TouchableOpacity
          style={styles.touchStyle}
          onPress={this.props.onPress}
        >
          <Icon name={iconName} size={30} color="#fff" />
          <Text
            style={styles.textStyle}
          >
            {this.props.type}
          </Text>
          <Text
            style={styles.subtitleStyle}
          >
            {this.props.subtitle}
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}




export default MenuBtn;