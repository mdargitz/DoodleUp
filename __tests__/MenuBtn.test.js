import React from 'react';
import MenuBtn from '../components/MenuBtn';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<MenuBtn />).toJSON();
  expect(tree).toMatchSnapshot();
});