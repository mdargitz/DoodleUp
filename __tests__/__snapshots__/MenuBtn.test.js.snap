// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3B3B58",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 30,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
        }
      }
    />
  </View>
</View>
`;
